### PR TITLE
Open-source the OpenStack DB migration scripts and workshop

### PR DESCRIPTION
**Pull request by 74togo**
Includes:
* Scripts for migrating various Openstack components' DB from Juno to Ocata version.
* Scripts for demonstrating these migrations within a VM.

### CODE REVIEW

**Review by Patchett — APPROVED** [review]
👍 👍 👍 👍 👍 👍 👍 👍

File: migration-scripts/migration-workshop/ocata/restore_dump.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# Copyright 2018, Oath Inc.
+# Licensed under the terms of the MIT license. See LICENSE file for terms.
+
+mysql -h192.168.33.10 -uroot -pkolla -e "show databases" | grep -v Database | grep -v mysql| grep -v information_schema| gawk '{print "drop database " $1 ";select sleep(0.1);"}' | mysql -h192.168.33.10 -uroot -pkolla
+mysql -h192.168.33.10 -uroot -pkolla < ocata_db_dump.sql
```
Comment (jimrollenhagen) [inline migration-scripts/migration-workshop/ocata/restore_dump.sh:6]: Does this IP need to be a variable? I guess not, as it's hardcoded in vagrant config, but something to think about.

File: migration-scripts/scripts/do_migration.sh [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    deactivate
+  done
+done
+
+# Remove old endpoints (chef deploy will re-populate these with correct values)
+mysql -u $db_user --password=$db_pass -h $db_host -e 'USE keystone; DELETE FROM endpoint;'
+mysql -u $db_user --password=$db_pass -h $db_host -e 'USE keystone; DELETE FROM service;'
+# Remove invalid role assignments (missing user)
+
+# Change legacy "config_drive" key to what is used upstream "configdrive"
```
Comment (jimrollenhagen) [inline migration-scripts/scripts/do_migration.sh]: We probably don't need this block for the workshop :)

**Review by jimrollenhagen — APPROVED** [review]
Looks good overall, a couple small things inline but I'd say this is good to go.

Is it worth mentioning in your workshop description that attendees should have a linux machine (whether laptop or server)? People are going to show up with a Mac and wonder why vagrant-libvirt doesn't work. Or maybe provide a virtualbox version of the Vagrantfile as well?

File: migration-scripts/migration-workshop/ocata/restore_dump.sh (hunk 1 above)
Comment (74togo) [reply migration-scripts/migration-workshop/ocata/restore_dump.sh:6]: yeah it's also the IP that is hard-coded to be in the endpoints/cells/compute nodes etc. 

basically just a simplification for the workshop, definitely not the way one should do things in production.

**Pull request merged**